include/node.hpp

```diff
@@ -708,6 +711,31 @@ class node : protected std::tuple<Arguments...> {
         // case sources: HW triggered vs. generating data per invocation (generators via Port::MIN)
         // case sinks: HW triggered vs. fixed-size consumer (may block/never finish for insufficient input data and fixed Port::MIN>0)
 
+        std::size_t n_samples_to_consume = ports_status.in_samples; // default stride == 0
+        if (stride != 0) {
```

Comment:
Could the stride evaluation be guarded with a constexpr-if? I.e. only being evaluated when the given tag is being set? I think we can re-use the existing one you already defined.

Reply:
Done, but with different tag `PerformStride`. One of the problems is that strides can be applied to sink nodes but decimation/interpolation not.